benchmarks/benchmark_serving.py

```diff
@@ -766,6 +778,10 @@ def main(args: argparse.Namespace):
                         default=None,
                         help="Path to the sharegpt/sonnet dataset. "
                         "Or the huggingface dataset ID if using HF dataset.")
+    parser.add_argument("--max-concurrency",
+                        type=int,
+                        default=None,
+                        help="Maximum number of concurrent requests.")
```

Comment:
Would be great if you can explain more on why setting `--max-concurrency` given that there is already `--request-rate` available. The main reason for me is to avoid request failure due to sending too many request to the inference engine, but I'm not sure.

Reply:
Perhaps to give some context - I think the main motivation here is that sometimes inference server are setup with max concurrency as a metric for autoscaling.

Currently on vLLM, we don't have a way to "reject" requests depending on the server load, so very often users set up this concurrency control at a higher level, thus it would be great if we can simulate this kind of setup in benchmark framework as well.

Reply:
I think it'd be great to add more information for this argument based on my reply to Cody's comment above if that makes sense.

Reply:
I expanded the help text. Let me know what you think!